src/main/java/net/dries007/tfc/common/blocks/devices/CalendarClockBlock.java

```diff
@@ -90,9 +109,34 @@ protected InteractionResult useWithoutItem(BlockState state, Level level, BlockP
             }
             else
             {
-                final BlockState blockstate = state.cycle(CLOCK_MONTH_MODE);
-                level.setBlock(pos, blockstate, 2);
-                level.gameEvent(GameEvent.BLOCK_CHANGE, pos, GameEvent.Context.of(player, blockstate));
+                BlockState blockState;
+                Direction direction = state.getValue(FACING);
+                switch (state.getValue(MODE)) {
+                    case Mode.HOUR:
+                    default: blockState = state.setValue(MODE, Mode.MONTH);
+                        break;
+                    case MONTH:
+                        if (level.getBlockState(pos
+                                .relative(direction.getOpposite()))
+                            .is(TFCTags.Blocks.CLOCK_READABLE))
```

Comment:
this is a lot of enters for no reason. Let's clean this up and get it on one line. Also use our `Helpers#isBlock`